sourcerer-esjc/src/main/java/org/elder/sourcerer/esjc/EventStoreEsjcEventRepository.java

```diff
@@ -426,10 +423,7 @@ private T normalizeEvent(final T rawEvent) {
         } catch (ExecutionException ex) {
             if (ex.getCause() instanceof EventStoreException) {
                 if (ex.getCause() instanceof WrongExpectedVersionException) {
-                    throw new UnexpectedVersionException(
-                            ex.getCause(),
-                            null,
-                            expectedVersion);
+                    throw new UnexpectedVersionException(ex.getCause(), expectedVersion);
                 } else if (ex.getCause() instanceof AccessDeniedException
```

Comment:
![MAJOR](https://sonarsource.github.io/sonar-github/severity-major.png 'Severity: MAJOR') Reduce the number of conditional operators (6) used in the expression (maximum allowed 3). [![rule](https://sonarsource.github.io/sonar-github/rule.png)](https://sonar.elder.org/coding_rules#rule_key=squid%3AS1067)